components/automate-ui/src/app/page-components/projects-filter-dropdown/projects-filter-dropdown.component.ts

```diff
@@ -33,27 +37,36 @@ export class ProjectsFilterDropdownComponent {
   // so they can be filtered while maintaining the actual options.
   filteredOptions: ProjectsFilterOption[] = [];
 
-  filterValue = '';
-
   optionsEdited = false;
 
   dropdownActive = false;
 
+
+  ngOnInit() {
+    this.projectsFilterService.filterValue$.subscribe((filterValue) => {
+      this.filteredOptions = this.filterOptions(filterValue);
+    });
+  }
```

Comment:
ngOnInit seemed like a logical place to put this at the time, but now I'm thinking that since this is global - are subscriptions expensive?  I'm also not currently unsubscribing at any point.

I'm wondering if maybe a better route is to kick off this subscription when the dropdown opens, and then unsubscribe when the dropdown closes... thoughts?

Reply:
I should also mention, this subscription is listening for the filterValue in state, which is what a user types in the filter input box, and then updates the projects dropdown list according to what is being filtered.

Reply:
You ask a good question ("are subscriptions expensive?") but that is really _two_ questions:

1. Is it expensive to create/destroy a subscription?
I would say "no" because best practice espoused by many articles is to create/destroy subscriptions as tightly as possible (create just-in-time and destroy as soon as you are done with it)

2. Is it expensive to maintain a subscription?
Not much in and of itself, as I imagine the rxjs design to monitor source Observables is rather efficient. But imagine if we never destroyed any subscriptions: We have a lot of "Observable overlap" in our code, since each component subscribes to what it needs, independent of what other components are doing. Particularly for those overlapping ones, every subscription will process the emitted Observable--even if it is not on a visible page and its result will never be used! (And sometimes leading to hard-to-find bugs.)

So while the global project filter is rather unique (it is created once and remains in existence for the life of the application) compared to almost every other component that is created and destroyed as things happen, we can still institute a tight create/destroy paradigm because of the lifecycle you described. And you have already led yourself to a good answer: open dropdown => subscribe, close dropdown => unsubscribe.

Reply:
interesting question! 💭 I think this is a case where we might not need a subscription.
[This article](https://medium.com/angular-in-depth/when-to-subscribe-a83332ae053) seems to discourage subscriptions unless absolutely necessary. It cites this [other article](https://medium.com/angular-in-depth/angular-question-rxjs-subscribe-vs-async-pipe-in-component-templates-c956c8c0c794) which argues that `async` in the template is the better option for using observable data, since it works with change detection out of the box and you don't have to remember to unsubscribe.